Makefile

```diff
@@ -83,53 +81,86 @@ install:
 	cp -Ra dbicdh "$(DESTDIR)"/usr/share/openqa/dbicdh
 
 
+.PHONY: check
+check: checkstyle test-with-database
+
 .PHONY: checkstyle
 checkstyle: test-shellcheck
 ifneq ($(CHECKSTYLE),0)
 	PERL5LIB=lib/perlcritic:$$PERL5LIB perlcritic lib
 endif
 
-.PHONY: test
-ifeq ($(TRAVIS),true)
-test: run-tests-within-container
-else
-test: checkstyle test-with-database
-endif
+.PHONY: phony_explicit
+phony_explicit: 
+
+.PHONY: test-with-db
+test-with-db: test-with-db-default
 
-.PHONY: test-unit-and-integration
-test-unit-and-integration:
-	OPENQA_CONFIG= prove ${PROVE_LIB_ARGS} ${PROVE_ARGS}
+test-with-db%: phony_explicit
+	@$(MAKE) $(subst test-with-db,test-with-database,$@)
 
 .PHONY: test-with-database
-test-with-database:
+test-with-database: test-with-database-default
+
+test-with-database%: phony_explicit
 	test -d $(TEST_PG_PATH) && (pg_ctl -D $(TEST_PG_PATH) -s status >&/dev/null || pg_ctl -D $(TEST_PG_PATH) -s start) || ./t/test_postgresql $(TEST_PG_PATH)
-	$(MAKE) test-unit-and-integration TEST_PG="DBI:Pg:dbname=openqa_test;host=$(TEST_PG_PATH)"
+	$(MAKE) $(subst -with-database,,$@) TEST_PG="DBI:Pg:dbname=openqa_test;host=$(TEST_PG_PATH)"
 	-pg_ctl -D $(TEST_PG_PATH) stop
 
-# prepares running the tests within Docker (eg. pulls os-autoinst) and then runs the tests considering
-# the test matrix environment variables
-# note: This is supposed to run within the Docker container unlike `launch-docker-to-run-tests-within`
-#       which launches the container.
-.PHONY: run-tests-within-container
-run-tests-within-container:
-	script/run-tests-within-container
+.PHONY: test
+test:
```

Comment:
As you insist on the new meaning of the target please make sure the documentation is adapted accordingly. Especially as you mentioned that it's currently (already) inconsistent.